Non-linear solver does not compile

In the [latest state of develop](4522fdba14aeb42b69e90050ae0a7f68063e4262) (as well as of master), the non-linear solver does not compile, reporting:

```
$ make 
Scanning dependencies of target nonlinear_elasticity
[ 50%] Building CXX object CMakeFiles/nonlinear_elasticity.dir/nonlinear_elasticity.cc.o
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc: In member function ‘void Nonlinear_Elasticity::Solid<dim, NumberType>::system_setup()’:
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:624:34: error: missing template arguments before ‘(’ token
     state_variables = std::vector({&total_displacement,
                                  ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc: In member function ‘void Nonlinear_Elasticity::Solid<dim, NumberType>::make_constraints(const int&)’:
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1359:50: error: ‘ZeroFunction’ was not declared in this scope
                                                  ZeroFunction<dim>(
                                                  ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1359:50: note: suggested alternative:
In file included from /home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1:0:
/usr/local/include/deal.II/base/function.h:510:9: note:   ‘dealii::Functions::ZeroFunction’
   class ZeroFunction : public ConstantFunction<dim, RangeNumberType>
         ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1366:50: error: ‘ZeroFunction’ was not declared in this scope
                                                  ZeroFunction<dim>(
                                                  ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1366:50: note: suggested alternative:
In file included from /home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1:0:
/usr/local/include/deal.II/base/function.h:510:9: note:   ‘dealii::Functions::ZeroFunction’
   class ZeroFunction : public ConstantFunction<dim, RangeNumberType>
         ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1383:13: error: ‘ZeroFunction’ was not declared in this scope
             ZeroFunction<dim>(n_components),
             ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1383:13: note: suggested alternative:
In file included from /home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1:0:
/usr/local/include/deal.II/base/function.h:510:9: note:   ‘dealii::Functions::ZeroFunction’
   class ZeroFunction : public ConstantFunction<dim, RangeNumberType>
         ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1390:13: error: ‘ZeroFunction’ was not declared in this scope
             ZeroFunction<dim>(n_components),
             ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1390:13: note: suggested alternative:
In file included from /home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1:0:
/usr/local/include/deal.II/base/function.h:510:9: note:   ‘dealii::Functions::ZeroFunction’
   class ZeroFunction : public ConstantFunction<dim, RangeNumberType>
         ^
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc: In instantiation of ‘void Nonlinear_Elasticity::Solid<dim, NumberType>::solve_nonlinear_timestep(dealii::BlockVector<double>&) [with int dim = 2; NumberType = double]’:
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:395:33:   required from ‘void Nonlinear_Elasticity::Solid<dim, NumberType>::run() [with int dim = 2; NumberType = double]’
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:1522:17:   required from here
/home/makish/github/dealii-adapter/nonlinear_elasticity/nonlinear_elasticity.cc:715:59: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
                error_residual_norm.u <= parameters.tol_f) ||
                                                           ^
CMakeFiles/nonlinear_elasticity.dir/build.make:62: recipe for target 'CMakeFiles/nonlinear_elasticity.dir/nonlinear_elasticity.cc.o' failed
make[2]: *** [CMakeFiles/nonlinear_elasticity.dir/nonlinear_elasticity.cc.o] Error 1
CMakeFiles/Makefile2:264: recipe for target 'CMakeFiles/nonlinear_elasticity.dir/all' failed
make[1]: *** [CMakeFiles/nonlinear_elasticity.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```
I am using deal.II 9.2 and GCC 5.5.0.

I will happily close this if it is only on my side and my ancient compiler.

--- a/nonlinear_elasticity/nonlinear_elasticity.cc
+++ b/nonlinear_elasticity/nonlinear_elasticity.cc
@@ -621,12 +621,12 @@ namespace Nonlinear_Elasticity
 
     // Alias: Container, which holds references for all time dependent variables
     // to enable a compact notation
-    state_variables = std::vector({&total_displacement,
-                                   &total_displacement_old,
-                                   &velocity,
-                                   &velocity_old,
-                                   &acceleration,
-                                   &acceleration_old});
+    state_variables = {&total_displacement,
+                       &total_displacement_old,
+                       &velocity,
+                       &velocity_old,
+                       &acceleration,
+                       &acceleration_old};
 
     setup_qph();
 
@@ -711,9 +711,10 @@ namespace Nonlinear_Elasticity
         // Check absolute errors for dynamic cases as well, since there might be
         // situations, with small or even no deformations in coupled setups
         if (newton_iteration > 0 &&
-              (error_update_norm.u <= parameters.tol_u &&
-               error_residual_norm.u <= parameters.tol_f) ||
-            (error_update.u <= 1e-15 && error_residual.u <= 5e-9))
+            ((error_update_norm.u <= parameters.tol_u ||
+              error_update.u <= 1e-15) &&
+             (error_residual_norm.u <= parameters.tol_f ||
+              error_residual.u <= 5e-9)))
           {
             std::cout << " CONVERGED! " << std::endl;
             print_conv_footer();
@@ -1356,14 +1357,14 @@ namespace Nonlinear_Elasticity
       if (apply_dirichlet_bc == true)
         VectorTools::interpolate_boundary_values(dof_handler_ref,
                                                  boundary_id,
-                                                 ZeroFunction<dim>(
+                                                 Functions::ZeroFunction<dim>(
                                                    n_components),
                                                  constraints,
                                                  fe.component_mask(u_fe));
       else
         VectorTools::interpolate_boundary_values(dof_handler_ref,
                                                  boundary_id,
-                                                 ZeroFunction<dim>(
+                                                 Functions::ZeroFunction<dim>(
                                                    n_components),
                                                  constraints,
                                                  fe.component_mask(u_fe));
@@ -1380,14 +1381,14 @@ namespace Nonlinear_Elasticity
           VectorTools::interpolate_boundary_values(
             dof_handler_ref,
             boundary_id,
-            ZeroFunction<dim>(n_components),
+            Functions::ZeroFunction<dim>(n_components),
             constraints,
             fe.component_mask(z_displacement));
         else
           VectorTools::interpolate_boundary_values(
             dof_handler_ref,
             boundary_id,
-            ZeroFunction<dim>(n_components),
+            Functions::ZeroFunction<dim>(n_components),
             constraints,
             fe.component_mask(z_displacement));
       }

